File: python/test/test_inner_product.py
import ThrustRTC as trtc



darr1 = trtc.device_vector_from_list([ 1.0, 2.0, 5.0], 'float')
darr2 = trtc.device_vector_from_list([ 4.0, 1.0, 5.0], 'float')
print (trtc.Inner_Product(darr1, darr2, trtc.DVFloat(0.0)))
print (trtc.Inner_Product(darr1, darr2, trtc.DVFloat(0.0), trtc.Plus(), trtc.Multiplies()))

File: java/test/test_inner_product.java
import java.util.*;
import JThrustRTC.*;

public class test_inner_product
{
	public static void main(String[] args) 
	{
        DVVector d_vec1 = new DVVector(new float[] { 1.0f, 2.0f, 5.0f });
        DVVector d_vec2 = new DVVector(new float[] { 4.0f, 1.0f, 5.0f });
       	System.out.println(TRTC.Inner_Product(d_vec1, d_vec2, new DVFloat(0.0f)));
        System.out.println(TRTC.Inner_Product(d_vec1, d_vec2, new DVFloat(0.0f), new Functor("Plus"), new Functor("Multiplies")));
	}
}